Math packages: givaro, fflas-ffpack, linbox

tornaria opened the pull request:
<!-- Mark items with [x] where applicable -->

#### General
- [x] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me

#### Notes

- This includes: `givaro`, `fflas-ffpack`, `linbox`.
- Everything is tested and check pass in x86_64, x86_64-musl, i686.
- Except `gf2x`, everything is nocross.
- I tried to make sure everything works with baseline x86_64 as defined [here](https://en.wikipedia.org/wiki/X86-64#Microarchitecture_Levels), essentially it means SSE2 is ok, but SSE3 is not.
- I added a general option `native`, which defaults to off. When turned on, it will let the libraries compile for the build machine which often results in significant improvements.

EDIT: split in smaller independent PR to ease review

dkwo commented:
Great job, I will test soon.

tornaria commented:
Nothing relevant changed: I cleaned up some leftover lines in `common/shlib`.

Chocimier commented on srcpkgs/givaro/template:
```diff
@@ -0,0 +1,38 @@
+
+if [ ! "$build_option_native" ]; then
+	configure_args="--enable-sse --enable-sse2
+	  --disable-sse3 --disable-ssse3 --disable-sse41 --disable-sse42
```
I would rather remove option and make in unconditional. People are free to use XBPS_CFLAGS="-march=native" to get optimized build. Works for this package, should work for others too.

ericonr replied:
I think the correct variable is `XBPS_TARGET_CFLAGS`? At least that's what's used in the `build-profiles` (the README there needs to be fixed, btw).

I agree that that's a better way of solving it, but it would be nice to make sure it's working. Also see #29732

Chocimier replied:
I mean XBPS_CFLAGS in etc/conf

tornaria replied:
It might work in this case (assuming the only effect of the --enable-sse* configure flags is to add -msse* to CFLAGS, I'd have to check).

However, some math packages have custom written asm for different architectures so selecting build option "native" will not necessarily be equivalent to just using `-march=native`.

Another point is that using a build option will make an annotation in the xbps, e.g.:
```
$ xq givaro | grep native
build-options: ~native 
```
vs.
```
$ xbps-query -S givaro | grep build
build-options: native 
```

dkwo commented:
@tornaria @ericonr Btw, don't forget this one ;)

tornaria commented:
@ericonr: rebased with only style changes. Tested with sage-9.4.rc2.

tornaria commented:
Latest changes:
- renamed `fflas_ffpack` to `fflas-ffpack` since that is the right package name (the tar.bz2 has the wrong name but the tar.gz has the correct one).
- added a patch from https://github.com/linbox-team/linbox/pull/274 which fixes a bug on i686; formerly we disabled a particular test for arch=i686 due to this, but now all tests pass.

tornaria commented:
@ericonr: let's do math stuff one at a time. I propose to start with this one which is the older one I've got. I think I applied all the style changes you told me in #30032.
